File: velolabs_repos/lattis_android/app/src/main/java/com/lattis/ellipse/domain/interactor/updatetrip/StopGetTripDetailsThreadIfApplicableUseCase.java
package com.lattis.ellipse.domain.interactor.updatetrip;

import com.lattis.ellipse.domain.executor.PostExecutionThread;
import com.lattis.ellipse.domain.executor.ThreadExecutor;
import com.lattis.ellipse.domain.interactor.UseCase;
import com.lattis.ellipse.presentation.ui.ride.service.util.ServiceAction;

import javax.inject.Inject;

import io.reactivex.Observable;

public class StopGetTripDetailsThreadIfApplicableUseCase extends UseCase<Boolean>

{
    private ServiceAction serviceAction;

    @Inject
    public StopGetTripDetailsThreadIfApplicableUseCase(ThreadExecutor threadExecutor,
                                 PostExecutionThread postExecutionThread,
                                 ServiceAction serviceAction) {
        super(threadExecutor, postExecutionThread);
        this.serviceAction = serviceAction;
    }

    @Override
    protected Observable<Boolean> buildUseCaseObservable() {
        return serviceAction.stopGetTripDetailsThreadIfApplicable();
    }
}

File: velolabs_repos/lattis_android_v2/domain/src/main/java/com/lattis/domain/usecase/activetripservice/StopGetTripDetailsThreadIfApplicableUseCase.kt
package com.lattis.domain.usecase.activetripservice

import com.lattis.domain.executor.PostExecutionThread
import com.lattis.domain.executor.ThreadExecutor
import com.lattis.domain.repository.ActiveTripRepository
import com.lattis.domain.usecase.base.UseCase
import io.reactivex.rxjava3.core.Observable
import javax.inject.Inject

class StopGetTripDetailsThreadIfApplicableUseCase @Inject constructor(
    threadExecutor: ThreadExecutor,
    postExecutionThread: PostExecutionThread,
    val activeTripRepository: ActiveTripRepository
) : UseCase<Boolean>(threadExecutor, postExecutionThread) {
    override fun buildUseCaseObservable(): Observable<Boolean> {
        return activeTripRepository.stopGetTripDetailsThreadIfApplicable()
    }
}